include/wil/registry_basic.h

```diff
@@ -1172,85 +1134,108 @@ namespace reg_view_details
     template <typename T, typename err_policy = err_exception_policy>
     static reg_view_t<T, err_policy> create_reg_view(T&& t)
     {
-        return reg_view_t<T, err_policy>(std::forward<T>(t));
+        return reg_view_t<T, err_policy>(::wistd::forward<T>(t));
     }
 
     // reg_view with the raw HKEY type is non-owning
     using reg_view_nothrow = reg_view_details::reg_view_t<HKEY, ::wil::err_returncode_policy>;
     using reg_view_unique_hkey_nothrow = reg_view_details::reg_view_t<::wil::unique_hkey, ::wil::err_returncode_policy>;
 #if defined(__WIL_WINREG_STL)
     using reg_view_shared_hkey_nothrow = reg_view_details::reg_view_t<::wil::shared_hkey, ::wil::err_returncode_policy>;
-#endif
+#endif // #if defined(__WIL_WINREG_STL)
 
-#ifdef WIL_ENABLE_EXCEPTIONS
+#if defined(WIL_ENABLE_EXCEPTIONS)
     // reg_view with the raw HKEY type is non-owning
     using reg_view = reg_view_details::reg_view_t<HKEY, ::wil::err_exception_policy>;
     using reg_view_unique_hkey = reg_view_details::reg_view_t<::wil::unique_hkey, ::wil::err_exception_policy>;
+#endif // #if defined(WIL_ENABLE_EXCEPTIONS)
 #if defined(__WIL_WINREG_STL)
     using reg_view_shared_hkey = reg_view_details::reg_view_t<::wil::shared_hkey, ::wil::err_exception_policy>;
-#endif
-#endif
+#endif // #if defined(__WIL_WINREG_STL)
 }
 
 // Registry Open* and Create* functions, both throwing and non-throwing
-
-#ifdef WIL_ENABLE_EXCEPTIONS
-inline ::wil::unique_hkey open_unique_key(HKEY key, _In_opt_ PCWSTR path, ::wil::registry::key_access access = ::wil::registry::key_access::read)
+#if defined(WIL_ENABLE_EXCEPTIONS)
+inline ::wil::unique_hkey open_unique_key(HKEY key, _In_opt_ PCWSTR path, ::wil::reg::key_access access = ::wil::reg::key_access::read)
 {
     const reg_view_details::reg_view regview{key};
     return regview.open_unique_key(path, access);
 }
 
-inline ::wil::unique_hkey create_unique_key(HKEY key, _In_ PCWSTR path, ::wil::registry::key_access access = ::wil::registry::key_access::read, _In_opt_ PCWSTR security_descriptor = nullptr)
+inline ::wil::unique_hkey create_unique_key(HKEY key, _In_ PCWSTR path, ::wil::reg::key_access access = ::wil::reg::key_access::read, _In_opt_ PCWSTR security_descriptor = nullptr)
 {
     const reg_view_details::reg_view regview{key};
     return regview.create_unique_key(path, access, security_descriptor);
 }
+#endif // #if defined(WIL_ENABLE_EXCEPTIONS)
+
+inline ::wil::unique_hkey try_open_unique_key(HKEY key, _In_opt_ PCWSTR path, ::wil::reg::key_access access = ::wil::reg::key_access::read)
+{
+    const reg_view_details::reg_view_nothrow regview{ key };
+    return regview.open_unique_key(path, access);

```

Comment:
Does this work? Shouldn't a nothrow call return an error code, not a `::wil::unique_hkey`?